core/templates/dev/head/components/explanation_editor_directive.html

```diff
@@ -0,0 +1,51 @@
+<div style="height: 100%;">
+  <div ng-if="!explanationEditorIsOpen"
+       style="height: 100%;"
+       ng-attr-title="<[isEditable ? 'Edit explanation' : '']>">
+    <div class="oppia-readonly-rule-tile"
+         ng-class="{'oppia-editable-section': isEditable}">
+      <div class="oppia-rule-preview oppia-transition-200">
+        <div class="oppia-click-to-start-editing"
+             ng-if="isEditable" ng-click="openExplanationEditor()">
+          <i class="material-icons oppia-editor-edit-icon pull-right"
+             title="Edit Explanation">&#xE254;
+          </i>
+        </div>
+        <strong>Explanation:</strong>
+        <span angular-html-bind="stateSolutionService.savedMemento.explanation"></span>
+      </div>
+    </div>
+  </div>
+
+  <div ng-if="isEditable && explanationEditorIsOpen">
+    <form role="form"
+          class="form-inline"
+          name="editSolutionForm"
+          ng-submit="saveThisExplanation()">
```

Comment:
Why do you need the ng-submit() here (or even the form, for that matter)? I don't see a button that would trigger the submit. The "Save" button below with the ng-click should be sufficient.